gravitee-am-identityprovider/gravitee-am-identityprovider-http/src/main/java/io/gravitee/am/identityprovider/http/authentication/HttpAuthenticationProvider.java

```diff
@@ -182,7 +121,135 @@
 
     @Override
     public Maybe<User> loadUserByUsername(String username) {
-        return Maybe.empty();
+        // prepare request
+        final HttpAuthResourcePathsConfiguration authResourceConfiguration = configuration.getAuthenticationResource().getPaths();
+        if (authResourceConfiguration == null) {
+            return Maybe.empty();
+        }
+        if (authResourceConfiguration.getLoadPreAuthUserResource() == null) {
+            return Maybe.empty();
+        }
+
+        final HttpResourceConfiguration readResourceConfiguration = authResourceConfiguration.getLoadPreAuthUserResource();
```

Comment:
IMHO Before going further, we should validate the configuration. We should at least provide an URI and a HTTP Method to process the request otherwise we should log a WARN and return empty.
Wdyt?

Reply:
FP